docs/pages/access-controls/idps/saml-guide.mdx

```diff
@@ -15,69 +15,98 @@ authenticate to external services.
 
 - (!docs/pages/includes/tctl.mdx!)
 - If you're new to SAML, consider reviewing our [SAML Identity Provider
-Reference](./saml-reference.mdx) before proceeding.
-
-## Example external application
-
-We'll be using [samltest.id](https://samltest.id/) to create a test consumer of
-Teleport's SAML identity provider.
-
-## Step 1/4. Configure a Teleport role with access to SAML service provider objects
-
-First, ensure you are logged into Teleport as a user that has permissions
-to read and modify `saml_idp_service_provider` objects. The default `editor` role
-has access to this already, but in case you are using a more customized configuration,
-create a role called `sp-manager.yaml` with the following contents:
-
+  Reference](./saml-reference.mdx) before proceeding.
+- User with preset `editor` role. We recommend creating a dedicated role to manage
+Service Provider.
+
+  <Admonition type="tip" title="Recommended">
+  <Details title="Create SAML IdP Service Provider role" closed>
+  First, ensure you are logged into Teleport as a user that has permissions
+  to read and modify `saml_idp_service_provider` objects. The default `editor` role
+  has access to this already, but in case you are using a more customized configuration,
+  create a role called `sp-manager.yaml` with the following contents:
+
+  ```yaml
+  kind: role
+  metadata:
+    name: sp-manager
+  spec:
+    allow:
+      rules:
+      - resources:
+        - saml_idp_service_provider
+        verbs:
+        - list
+        - create
+        - read
+        - update
+        - delete
+  version: v7
+  ```
+
+  Create the role with `tctl`:
+
+  ```code
+  $ tctl create sp-manager.yaml
+  role 'saml-idp-service-provider-manager' has been created
+  ```
+
+  Add the role to your user. Replace `teleport-admin` with your user name:
+
+  ```code
+  $ tctl users update --set-roles $(tctl get users/<Var name="teleport-admin"/> --format=json | \
+  jq -r '.[].spec.roles | join(",")'), sp-manager <Var name="teleport-admin"/>
+  User teleport-admin has been updated:
+    New roles: auditor,editor,access,sp-manager
+  ```
+  </Details>
+  </Admonition>
+- Test SAML application, also known as SAML Service Provider (SP). For this guide, we'll be using 
```

Comment:
This is also a bit tricky. This bullet actually corresponds to the list above, where the previous list item is 30 lines away. Can you think of ways to restructure this so we don't break the user's context? By the time I got to this point I forgot that I was still reading preqrequisites.

Reply:
Hmmm, I get your concern, I assume that is the case while reading directly from the `mdx` file and not through rendered page?  Though the block above is now moved to end (https://github.com/gravitational/teleport/pull/36467/commits/20806d4d4e6d30df09e5ec9f756d329f988623b7) so should address your concern here.